### PR TITLE
[TR] PIM-4615: Rework drafts to apply changes on backend

### PR DESCRIPTION
**Pull request by phaseinducer**
https://akeneo.atlassian.net/browse/PIM-4615

### CODE REVIEW

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/product-manager.js [hunk 1]
```diff
@@ -25,16 +25,9 @@ define([
                             return this.generateMissing(product);
                         }, this))
                         .then(function (product) {
-                            var promises = [];
-                            mediator.trigger('product:action:post_fetch', {
-                                promises: promises,
-                                product: product,
-                                originalProduct: $.extend(true, {}, product)
-                            });
+                            mediator.trigger('product:action:post_fetch', product);
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/manager/product-manager.js]: :heart_eyes:

File: src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js [hunk 2]
```diff
@@ -94,20 +94,16 @@ define(
...
                 var field = event.field;
-                event.promises.push(
-                    this.canBeCopied(field)
-                        .then(_.bind(function (canBeCopied) {
-                            if (canBeCopied && this.copying) {
-                                field.addElement('comparison', this.code, this.getCopyField(field));
-                            }
-                        }, this))
-                );
+                if (this.copying && this.canBeCopied(field)) {
```
Comment (sumbobyboys) [inline src/Pim/Bundle/EnrichBundle/Resources/public/js/product/form/attributes/copy.js]: :camel:

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php [hunk 3]
```diff
@@ -177,6 +184,7 @@ public function postAction(Request $request, $id)
 
         if (0 === $violations->count()) {
             $this->productSaver->save($product);
+            $this->objectManager->refresh($product);
```
Comment (jjanvier) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: possible to put it where we handle the draft ?

**Comment by jjanvier** [comment]
GTM apart from my comment

File: src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php (hunk 3 above)
Comment (phaseinducer) [inline src/Pim/Bundle/EnrichBundle/Controller/Rest/ProductController.php]: Much better in the delegating saver indeed :+1:

**Pull request merged**